ModelCatalogueGenomicsPlugin/src/groovy/org/modelcatalogue/gel/export/RareDiseaseChangeLogXlsExporter.groovy

```diff
@@ -338,101 +334,152 @@ abstract class RareDiseaseChangeLogXlsExporter extends AbstractChangeLogGenerato
         }
         deepElementValue
     }
 
-    //probably needs more protection
+
     private String extractChangedElement(String elemName) {
-        if (elemName && elemName.length() > 15) {   //some length indicating it might be a json string
+        if (mightBeJSON(elemName)) {
             def jsonMap
             def jsonSlurper = new JsonSlurper()
             try {
                 jsonMap = jsonSlurper.parseText(elemName)
             } catch (Exception e) {
                 log.error "Error attempting json parse of $elemName"
+                log.error ExceptionUtils.getStackTrace(e)
                 return elemName
             }
-            if (jsonMap && jsonMap.destination) {
-
-                String destination = jsonMap.destination.replaceAll(/[{}]/, '')
-                Map<String, String> destinationMap = new HashMap<String, String>().withDefault { ['', ''] }
-                def csvTokens = destination.split(',')
-                csvTokens.each { csvToken ->
-                    String[] tokens = csvToken.split('=')
-                    if (tokens.length < 2) {
-                        destinationMap.put(tokens[0], tokens[0])
+
+
+            try {
+                if (jsonMap && jsonMap.destination) {
+
+                    String destination = jsonMap.destination.replaceAll(/[{}]/, '')
+                    Map<String, String> destinationMap = new HashMap<String, String>().withDefault { ['', ''] }
+                    def csvTokens = destination.split(',')
+                    csvTokens.each { csvToken ->
+                        String[] tokens = csvToken.split('=')
+                        if (tokens.length < 2) {
+                            destinationMap.put(tokens[0], tokens[0])
+                        } else {
+                            destinationMap.put(tokens[0].trim(), tokens[1])
+                        }
+                    }
+                    if (destinationMap.classifiedName && destinationMap.classifiedName instanceof String) {
+                        elemName = "${destinationMap.classifiedName ?: ''} (${destinationMap.semanticVersion ?: ''})"
                     } else {
-                        destinationMap.put(tokens[0].trim(), tokens[1])
+                        elemName = ''
                     }
                 }
-                if(destinationMap.classifiedName && destinationMap.classifiedName instanceof String) {
-                    elemName = "${destinationMap.classifiedName ?: ''} (${destinationMap.semanticVersion ?: ''})"
-                } else {
-                    elemName = ''
-                }
+            } catch (Exception e) {
+                log.error "Error attempting csv token parse of $elemName"
+                log.error ExceptionUtils.getStackTrace(e)
+                return elemName
             }
         }
         elemName
     }
 
 
+    //is it potentially useful json? unlikely if less than 15 chars
+    public static boolean mightBeJSON(String jsonStr) {
```

Comment:
this might do the better job - https://github.com/MetadataRegistry/ModelCataloguePlugin/blob/e95528bf9e508f2326ed0304784e239ed3919158/ModelCatalogueCorePlugin/src/groovy/org/modelcatalogue/core/audit/LoggingAuditor.groovy#L426-L426